### PR TITLE
Gwt ide as sidecar Docker image

### LINKED ISSUE #12172: Package GWT IDE as Che Editor
### Description
We need to package an release GWT IDE as a sidecar and contribute it as a CheEditor in https://github.com/eclipse/che-plugin-registry

### PR DESCRIPTION
**Pull request by skabashnyuk**
### What does this PR do?
Gwt ide as sidecar Docker image

### What issues does this PR fix or reference?
Fixes https://github.com/eclipse/che/issues/12172

#### Release Notes
n/a

#### Docs PR
n/a

### CODE REVIEW

File: dockerfiles/gwt-ide/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    && echo "%wheel ALL=(ALL) NOPASSWD: ALL" >> /etc/sudoers \
+    # Create /projects for Che
+    && mkdir /projects \
+    && touch /var/log/supervisord.log \
+    && cat /etc/passwd | sed s#root:x.*#root:x:\${USER_ID}:\${GROUP_ID}::\${HOME}:/bin/bash#g > ${HOME}/passwd.template \
+    && cat /etc/group | sed s#root:x:0:#root:x:0:0,\${USER_ID}:#g > ${HOME}/group.template \
+    # Cleanup tmp folder
+    && rm -rf /tmp/* \
+    # Change permissions to allow editing of files for openshift user
+    && for f in "${HOME}" "/etc/passwd" "/etc/group /var/log/ /projects"; do\
```
Comment (benoitf) [inline dockerfiles/gwt-ide/Dockerfile]: FYI it seems the quotes are strange on this line, etc/group, /var/log and /projects being in the same 'group'

**Comment by benoitf** [comment]
@skabashnyuk do you see e2e tests that we could add on top of this image ?

File: dockerfiles/gwt-ide/Dockerfile (hunk 1 above)
Comment (skabashnyuk) [reply dockerfiles/gwt-ide/Dockerfile]: Can you explain what you want me to change?

**Comment by skabashnyuk** [comment]
>@skabashnyuk do you see e2e tests that we could add on top of this image?

At this moment - no, without big community request for it I don't think it would be reasonable to invest time on it.

**Review by benoitf — APPROVED** [review]

File: dockerfiles/gwt-ide/traefik.toml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+# Required
+#
+#address = ":8086"
+
+
+
+[file]
+
+[backends]
+  [backends.backend1]
```
Comment (sleshchenko) [inline dockerfiles/gwt-ide/traefik.toml]: Please consider using meaningful names for backends, like `apiBackend`, `ideBackend`

**Review by sleshchenko — APPROVED** [review]
LGTM

**Comment by skabashnyuk** [comment]
ci-test

File: dockerfiles/gwt-ide/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    # Create /projects for Che
+    && mkdir /projects \
+    && touch /var/log/supervisord.log \
+    && cat /etc/passwd | sed s#root:x.*#root:x:\${USER_ID}:\${GROUP_ID}::\${HOME}:/bin/bash#g > ${HOME}/passwd.template \
+    && cat /etc/group | sed s#root:x:0:#root:x:0:0,\${USER_ID}:#g > ${HOME}/group.template \
+    # Cleanup tmp folder
+    && rm -rf /tmp/* \
+    # Change permissions to allow editing of files for openshift user
+     && for f in "${HOME}" "/etc/passwd" "/etc/group" "/var/log/" "/projects"; do\
+               sudo chgrp -R 0 ${f} && \
```
Comment (benoitf) [inline dockerfiles/gwt-ide/Dockerfile]: probably here we could avoid to use sudo as we're root (and then sudo package should not be installed)

**Comment by riuvshin** [comment]
Results of automated E2E tests of Eclipse Che Multiuser on OCP:
[Build details](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/1333/)
[Test report](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/1333//Selenium_20tests_20report/)
docker image: eclipseche/che-server:12201
https://github.com/orgs/eclipse/teams/eclipse-che-qa please check this report.

**Review by vparfonov — APPROVED** [review]

**Pull request merged**